### ISSUE TITLE
[accessibility issue] Input component placeholder text doesn't meet WCAG AA contrast requirements (light mode only)

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#79571]
**Steps to reproduce**:
Go to any Input component with placeholder text in light mode (or the docs site: https://developers.grafana.com/ui/latest/index.html?path=/story/forms-input--simple). [Use a color contrast site](https://www.siegemedia.com/contrast-ratio#rgba%2836%2C%2041%2C%2046%2C%200.5%29-on-%23fff) to establish the contrast is only 3.04. Minimum is 4.5.

**Expected Result**
Contrast ratio is at least 4.5:1.

A good example is the command palette at the top of every page. The "Search or jump to..." placeholder text.

<img width="354" alt="The Grafana Command Palette in light mode." src="https://github.com/grafana/grafana/assets/6906380/31cf73f3-c9fe-4a07-bc20-47b3723ee5fd">
